src/components/date-picker/DatePicker.tsx

```diff
@@ -443,17 +598,27 @@ class DatePicker extends React.Component<DatePickerProps> {
 
         const resinTargetAttr = resinTarget ? { [RESIN_TAG_TARGET]: resinTarget } : {};
 
-        const valueAttr = isTextInputAllowed
-            ? { defaultValue: this.formatDisplay(value) }
-            : { value: this.formatDisplay(value) };
+        let valueAttr;
+        if (isAccessible) {
+            valueAttr = { defaultValue: this.formatDisplayDateType(value) };
+        } else if (isTextInputAllowed) {
+            valueAttr = { defaultValue: this.formatDisplay(value) };
+        } else {
+            valueAttr = { value: this.formatDisplay(value) };
+        }
+        let onChangeAttr;
+        if (isAccessible && this.canUseDateInputType) {
+            onChangeAttr = { onChange: this.handleOnChange };
+        } else if (isTextInputAllowed || (isAccessible && !this.canUseDateInputType)) {
+            onChangeAttr = {};
+        } else {
+            onChangeAttr = { onChange: noop };
+        }
+        // "name" prop is required for pattern validation to be surfaced on form submit. See components/form-elements/form/Form.js
+        const additionalAttrs =
+            isAccessible && !this.canUseDateInputType ? { name, pattern: '\\d{4}-\\d{2}-\\d{2}' } : {};
```

Comment:
Is there anything we should do when we are in text fallback mode to help make sure the date is not in ddmm order instead of mmdd? Such as validate the 1-31 and 1-12 ranges?